STYLEGUIDE.md

```diff
@@ -31,3 +33,14 @@
 - Run the linter and follow all rules defined in .eslintrc
 - Never use absolute paths in code. The app should be able to run behind a proxy under an arbitrary path.
 - TESTS: Should follow a similar "test tables" convention as used in Go where applicable.
+
+#### Type Safety
+- Use specific Kubernetes resource types instead of generic `K8sResourceCommon` when possible
+- Avoid excessive use of `any` types
+- Avoid type assertions with `as any`
+- Use optional chaining for safe property access
```

Comment:
I'd also say that fallback behavior should make sense. (For instance, `fetch(obj?.url)` doesn't make sense even if it avoids a runtime error.)